src/main/java/org/opensearch/security/ssl/transport/SecuritySSLRequestHandler.java

```diff
@@ -86,6 +89,11 @@ public final void messageReceived(T request, TransportChannel channel, Task task
 
         ThreadContext threadContext = getThreadContext();
 
+        String channelType = channel.getChannelType();
```

Comment:
> When PA [instantiates a PATransportChannel](https://github.com/opensearch-project/performance-analyzer/blob/main/src/main/java/org/opensearch/performanceanalyzer/transport/PerformanceAnalyzerTransportRequestHandler.java#L80-L81) it does not copy the version from the original [TransportChannel](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/transport/TransportChannel.java#L61-L66). Since it doesn't copy the version from the original, it will go with the default from TransportChannel which is the value of Version.CURRENT on the receiving node.

This seems like a bug in PA - I'm not sure I understand why we need a change in the security code?

Reply:
IMO The primary benefit of this PR is that it adds a test for the [getInnerChannel logic](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/ssl/transport/SecuritySSLRequestHandler.java#L191-L203). The tests in this PR will enforce that operations on a channel within the security plugin should only be done after extracting the inner channel if the channel is wrapped.

Reply:
+1 - fixing here also ensures it preserves the invariant of always acting on extracted channel and protects from other plugin bugs causing similar impact (though the underlying incorrect wrapping also need to be fixed).

Reply:
Just for information, this issue is also appearing when there is not PA enabled on the cluster. I faced the same issue while upgrading from 1.3 to 2.11.1